lib/root_component.dart

```diff
@@ -1,13 +1,45 @@
 import 'package:angular/angular.dart';
+import 'package:ng_bootstrap/ng_bootstrap.dart';
 
+import 'src/credential_service.dart';
+import 'src/excel.dart';
 import 'src/query_component.dart';
-import 'src/credential_component.dart';
 
 @Component(
   selector: 'application-root',
   templateUrl: 'root_component.html',
-  directives: [CredentialComponent, QueryComponent],
+  directives: [bsTabsxDirectives, coreDirectives, QueryComponent],
+  providers: [ClassProvider(CredentialService), ClassProvider(ExcelDart)],
 )
-class RootComponent {
+class RootComponent implements OnInit {
+  // Names used in html.
   final title = 'Display & Video 360';
+  final welcomeTitle = 'Welcome!';
+  final welcomeMessage =
+      'Discover what DV360 Excel Add-in can do for you today';
+  final sideloadMessage = 'Please sideload the add-in to continue';
+  final queryBuilderTabName = 'Query Builder';
+  final rulesBuilderTabName = 'Rules Builder';
+  final signOnButtonName = 'GET STARTED';
+  final signOutButtonName = 'Sign out';
+
+  // The current sign-in status.
+  bool isUserValidated;
+
+  // The current environment status.
```

Comment:
im missing where user validated and excelEnvironment are set.

Reply:
[here](https://github.com/googleinterns/dv360-excel-plugin/pull/80/files/17ba7638638ab0471dc29e4f421b110acecce154#diff-0ddd5d14fdbdacb8aa08fb8299ded715R38-R44)

Reply:
aha there it is